eclair-core/src/test/scala/fr/acinq/eclair/message/PostmanSpec.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.message
+
+import akka.actor.testkit.typed.scaladsl.{ScalaTestWithActorTestKit, TestProbe}
+import akka.actor.typed.ActorRef
+import akka.actor.typed.eventstream.EventStream
+import com.typesafe.config.ConfigFactory
+import fr.acinq.bitcoin.ByteVector32
+import fr.acinq.eclair.io.Switchboard.RelayMessage
+import fr.acinq.eclair.message.OnionMessages.ReceiveMessage
+import fr.acinq.eclair.message.Postman._
+import fr.acinq.eclair.wire.protocol.MessageOnion.FinalPayload
+import fr.acinq.eclair.wire.protocol.OnionMessagePayloadTlv.{EncryptedData, ReplyPath}
+import fr.acinq.eclair.wire.protocol.{GenericTlv, TlvStream}
+import fr.acinq.eclair.{UInt64, randomBytes32, randomKey}
+import org.scalatest.Outcome
+import org.scalatest.funsuite.FixtureAnyFunSuiteLike
+import scodec.bits.HexStringSyntax
+
+import scala.concurrent.duration._
+
+class PostmanSpec extends ScalaTestWithActorTestKit(ConfigFactory.load("application")) with FixtureAnyFunSuiteLike {
```

Comment:
This file deserves a few more tests IMO:

- send message without pathId that fails to be relayed -> sender should receive a failure
- send message without pathId that is relayed -> sender should receive a success
- send message with pathId that fails to be relayed -> sender should receive a failure
- send message with pathId that is relayed (postman receives the event from switchboard) but then times out -> sender should only receive the timeout error, not the switchboard "sent" event

Reply:
Done